snuba/request/validation.py

```diff
@@ -105,7 +106,13 @@ def build_request(
             record_error_building_request(timer, referrer)
             raise exception
 
-        span.set_data("snuba_query", request.body)
+        parsed_and_raw = {
+            "Parsed": repr(query).split("\n"),
+            "Raw": textwrap.wrap(repr(request.body), 100, break_long_words=False),
+        }
```

Comment:
We do not need an additional nesting level for the `snuba_query` item in the trace. 
You can call `set_data` multiple times with different keys. 
```
span.set_data('parsed_query': repr(......))
span.set_data('raw_query': textwrap.wrap(......)
```